server/boards/queries.ts

```diff
@@ -427,6 +427,19 @@ export const getBoardRoles = async ({
   return boardRoles;
 };
 
+export const createBoard = async (metadata: Partial<BoardByExternalId>) => {
```

Comment:
Instead of using `metadata: Partial<BoardByExternalId>`, each of the variables being passed to the function should be explicitly declared by the function.

As such, it might look something like this:
```suggestion
export const createBoard = async ({
  slug,
  realmExternalId,
  boardCategoryId,
  avatarUrl,
  tagline,
  settings
}: {
  slug: string;
  realmExternalId: string;
  boardCategoryId: string;
  avatarUrl: string;
  tagline: string;
  settings: object
}) => {
```